matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/database/query/ReadQueries.kt

```diff
@@ -94,7 +94,7 @@ internal fun isReadMarkerMoreRecent(realmConfiguration: RealmConfiguration,
             val eventToCheckIndex = eventToCheck?.displayIndex ?: Int.MAX_VALUE
             eventToCheckIndex <= readMarkerIndex
         } else {
-            eventToCheckChunk?.isLastForward == false
+            eventToCheckChunk?.isLastForward == false && readMarkerChunk?.isLastForward == true
```

Comment:
Good catch, but I think you should instead check for `readMarkerChunk.isMoreRecentThan(eventToCheckChunk)`

Reply:
Makes sense, thanks!

Reply:
Hm, my first quick test using `eventToCheckChunk != null && readMarkerChunk?.isMoreRecentThan(eventToCheckChunk) == true` doesn't work as intended. Will probably try more tomorrow.

Reply:
Found the issue! There is another bug in `ChunkEntityHelper.isMoreRecentThan()`, pushing the fix as additional commit to this PR.